broker.go

```diff
@@ -0,0 +1,726 @@
+package kafka
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/husio/kafka/proto"
```

Comment:
Can we create a separate PR to create the sub-package first, then point this one to the main repo rather than a fork?